### PR TITLE
feat: new user log in/sign up flow [ch630]

### PR DESCRIPTION
**Pull request by Jarrydcrawford**
- Rejig the /log-in and /sign-up routes and remove old create-account routes.
- Preparation for new screens.
- ~One component in a file, why not default export? 🤷‍♂~ @harrygr says no.

### CODE REVIEW

**Comment by harrygr** [comment]
> One component in a file, why not default export? 🤷‍♂

There's a number of reasons. The biggest for me is IDE support; it's way easier for your editor to detect and auto-import named exports. It also prevents naming some default import the wrong thing.

I think we should aim only to use named exports and ban default exports altogether.

File: src/App.tsx [hunk 1]
```diff
@@ -9,12 +9,12 @@ import { Loader } from './components/LoadingContainer';
...
 
 const UnauthenticatedApp = React.lazy(() =>
-  import('./UnauthenticatedApp').then(({ UnauthenticatedApp }) => ({
+  import('./UnauthenticatedApp').then(({ default: UnauthenticatedApp }) => ({
     default: UnauthenticatedApp,
   })),
 );
 const AuthenticatedApp = React.lazy(() =>
-  import('./AuthenticatedApp').then(({ AuthenticatedApp }) => ({
+  import('./AuthenticatedApp').then(({ default: AuthenticatedApp }) => ({
```
Comment (harrygr) [inline src/App.tsx]: Would prefer to leave as named exports. Also, if switching to default exports this `then` block doesn't do anything - it's literally returning its parameter.

File: src/AuthenticatedApp.tsx [hunk 2]
```diff
@@ -52,3 +52,5 @@ export const AuthenticatedApp: React.FC<Props> = () => {
     </Switch>
   );
 };
+
+export default AuthenticatedApp;
```
Comment (harrygr) [inline src/AuthenticatedApp.tsx]: 🙅‍♂

File: src/pages/SignUpOrLogIn/index.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { Redirect, Switch, Route, RouteComponentProps } from 'react-router';
+import { AuthLayoutWithAction } from '../../layouts/AuthLayout';
+import SignUp from '../Auth/SignUp';
+import LogIn from '../Auth/LogIn';
+
+interface Props {
+  isLoggedIn: boolean;
+}
+
+const SignUpOrLogIn: React.FC<Props & RouteComponentProps> = props => {
```
Comment (harrygr) [inline src/pages/SignUpOrLogIn/index.tsx]: What's the point in this component?

1. It should never be rendered if logged in (only the `UnauthenticatedApp` renders it), so no need for the `isLoggedIn` prop
2. Why have a subrouter? Just have the routes in the `UnauthenticatedApp` and wrap each page with the `AuthLayoutWithAction`

Comment (Jarrydcrawford) [reply src/pages/SignUpOrLogIn/index.tsx]: okie dokie!

File: src/App.tsx (hunk 1 above)
Comment (Jarrydcrawford) [reply src/App.tsx]: sure

File: src/UnauthenticatedApp.tsx [hunk 4]
```diff
@@ -1,32 +1,43 @@
...
+export const UnauthenticatedApp: React.FC = () => {
   return (
     <Switch>
-      <Route path="/" render={() => <Intro isLoggedIn={false} />} exact />
-      <Route component={CreateAccount} exact={true} path="/create-account" />
-      <Route component={Register} exact={true} path="/register" />
-      <Route component={Login} exact={true} path="/login" />
+      <Route
+        render={() => (
+          <AuthLayoutWithAction action="log-in">
```
Comment (harrygr) [inline src/UnauthenticatedApp.tsx]: Can't you just put the `AuthLayoutWithAction` inside the `Signup` and `Login` components?

Comment (Jarrydcrawford) [reply src/UnauthenticatedApp.tsx]: 🤷‍♂ sure

**Pull request merged**